include/svs/index/vamana/iterator.h

```diff
@@ -244,15 +208,15 @@ class BatchIterator {
     /// @copydoc begin()
     const_iterator cend() const { return results_.cend(); }
 
-    /// @brief Return a span over the current batch of neighbors.
-    ///
-    /// The return span will be invalidated by calls to ``next()``.
+    /// @brief Returns a span over the current batch of neighbors.
+    /// The span is invalidated by calls to `next(...)`.
     std::span<const value_type> contents() const { return lib::as_const_span(results_); }
 
-    /// @brief Return the number of buffered results.
+    /// @brief Returns the number of buffered results.
     size_t size() const { return results_.size(); }
 
-    /// @brief Return the current batch number contained in the buffer.
+    /// @brief Reurn the current batch nu contained in the buffermber contained in the
+    /// buffer.
```

Comment:
Have you considered a more explicit name for this method, such as `batch_number` or something similar? When I see just `batch` I think of the data in the buffer itself, not an iteration number.